### PR TITLE
Share `settings.gradle` plugins between composite builds

### PR DESCRIPTION
**Pull request by fondesa**
This PR shares the configuration of our JDK toolchain resolver moving the version in our Version Catalog and creating a composite build which provides a `Settings` plugin (not a `Project` ones).

_Why `SettingsPlugin` is in Groovy rather than Kotlin?_
We could have it in Kotlin but this means that we need to apply our Kotlin config to this new module from scratch, since it can't access any of our Kotlin shared config.
To do something simple like this, Groovy is way easier to maintain since it's shipped with Gradle so we don't need any custom config.

_How can we access the Version Catalog now but we couldn't before?_"
This happens because the Version Catalog is just a file which is provided **after** the settings.gradle plugins are resolved.
Now, this plugin is applied inside **settings.gradle** of the other projects ("aditogether" and "build-tools") but the Version Catalog is accessed only inside our **build.gradle** files.
TL;DR: Version Catalog is still not accessible by plugins in **settings.gradle** but now there's no need to since it's only references in **build.gradle** files.

Closes: https://github.com/androiddevelopersitalia/aditogether/issues/65

### CODE REVIEW

**Review by 4face-studi0 — APPROVED** [review]
Never seen such sorcery 😃
